File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/UseArrayEmpty.cs
namespace Tests.Diagnostics
{
    public class UseArrayEmpty
    {
        public const int Zero = 0;

        public UseArrayEmpty()
        {
            var array = new int[0]; // Noncompliant {{Declare this empty array using Array.Empty<T>.}}
            //          ^^^^^^^^^^

            var dbl = new int[00]; // Noncompliant
            var other = new int[] { }; // Noncompliant
            var comment = new int[] { /* comment */ }; // Noncompliant
            var dynamic = new dynamic[] { };// Noncompliant
            var cnst = new int[Zero]; // Noncompliant

            var array_1 = new int[1]; // Compliant
            var other_1 = new[] { 17 }; // Compliant
            var multi = new int[0, 0]; // Compliant

            Arguments(new int[42] { }); // Compliant
            Arguments(new int[0]); // Noncompliant
        }

        public void Arguments(int[] arguments)
        {
        }
    }
}

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/UseArrayEmpty.vb
Option Strict Off

Namespace Tests.Diagnostics

    Public Class UseArrayEmpty

        Public Const MinOne As Integer = -1

        Public Sub Arrays()

            Dim eplicit(-1) As Integer ' Noncompliant {{Declare this empty array using Array.Empty(Of T).}}
            '   ^^^^^^^^^^^^^^^^^^^^^^

            Dim implicit1 As Integer() = New Integer() {} ' Noncompliant
            Dim implicit2 As Integer() = {} ' // Noncompliant
            Dim implicit3() As Integer = {} ' Noncompliant
            Dim cnst As Integer(MinOne) ' Nonecompliant

            Dim dynamic = {} ' Noncompliant

            Dim array_1(0) As Integer ' Compliant
            Dim other_1() As Integer = {1} ' Compliant

            Arguments({42}) ' Compliant
            Arguments({}) ' Noncompliant

        End Sub

        Public Sub Arguments(arguments As Integer())
        End Sub

    End Class

End Namespace
